process/block/preprocess/gasComputation.go

```diff
@@ -168,7 +168,7 @@ func (gc *gasComputation) ComputeGasConsumedByTx(
 	txGasLimitConsumption := gc.economicsFee.ComputeGasLimit(txHandler)
 
 	if core.IsSmartContractAddress(txHandler.GetRcvAddr()) {
-		if txSenderShardId != txReceiverShardId {
+		if txSenderShardId != txReceiverShardId && txGasLimitConsumption < txHandler.GetGasLimit() {
```

Comment:
The problem was when gas was computed for a cross shard tx calling a SC or for a SCR?

Reply:
yes. in that case the SCR can have 0 gasLimit even if it contains some data - it was paid before from initial TX.

Reply:
In this case it is not better to add next check above, in line 167 for example ?
     if core.IsSmartContractAddress(txHandler.GetSndAddr()) {
          return uint64(0), uint64(0), nil
     }

Reply:
no. it should be with the given gaslimit.